### ISSUE TITLE
Configuration to automatically delete `pact` directory when re-running contract tests

### ISSUE THREAD

**Issue by dsinecos** [pact-foundation/pact-go#173]
## Software versions

* **OS**: Ubuntu 20.04
* **Consumer Pact library**: pact-go v2.0.0-20210621102432-26b32fd1552a
* **Provider Pact library**: -
* **Golang Version**: go1.16.2 linux/amd64
* **Golang environment**: -

## Expected behaviour

When running contract tests using `go test`, presently I need to either manually delete the pact directory or configure a `Before` hook to delete the pact directory. If I don't do this certain changes will not be overwritten when re-running the pact tests.

For eg. when I change the text inside matchers eg. `pact.S("Original text")` to `pact.S("Changed text")`. It doesn't get updated in the generated pact files. I need to delete the pact folder and rerun the tests for these changes to take effect.

## Actual behaviour

Based on a conversation with Matt in slack ( [Refer for conversation](https://pact-foundation.slack.com/archives/C9UTHTFFB/p1627278370142300) ) the functionality to automatically delete the pact directory exists in the shared core. The configuration will have to be exposed.

This configuration will automatically delete the pact directory on rerunning the contract tests.

## Steps to reproduce

-

## Relevent log files

-

**Comment by mefellows** [comment]
Thanks for this, I'll look at this on my next OSS day.

**Comment by YOU54F** [comment]
So I think the original defect was resolved in https://github.com/pact-foundation/pact-reference/issues/77 by commit https://github.com/pact-foundation/pact-reference/commit/3acf4376bc475e0b1a2acaa08c2b1505ad36d4b4

I can't see anything in the shared core for deleting files prior to a test run

**Comment by mefellows** [comment]
Yeah it might be, I can't remember what I was thinking so far back. The core doesn't know when the testing starts or finishes - each interaction is effectively tested on its own (it gets its own mock server, for example). Probably any cleaning would need to happen in the client or by the user (current recommendation).

**Comment by YOU54F** [comment]
So you can do a pre-test in your Go code. 

Here I am scoping it with a consumer build tag, so it runs before our consumer tests, but not before our provider tagged tests.

```golang
//go:build consumer
// +build consumer

package main

import (
	"os"
	"testing"
)

func TestMain(m *testing.M) {
	// Remove specific files before running tests
	if err := os.RemoveAll("pacts"); err != nil {
		panic(err)
	}

	// Run tests
	code := m.Run()

	// Exit with the test result code
	os.Exit(code)
}
```

This will clear pacts prior to a run

```sh
go test -v -tags=consumer -count=1 github.com/pact-foundation/pact-go/v2/examples/...
```

This won't

```sh
go test -v -tags=provider -count=1 github.com/pact-foundation/pact-go/v2/examples/...
```
